### PR TITLE
Revert "chore: remove reasoning example to safe my credit card (#116)"

### PR DESCRIPTION
**Pull request by OskarStark**
This reverts commit 4a48bc4f875f9de1aaa9e1d1016980b677eec7b1 and added checks via ENV var

Feel free to close

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -24,3 +24,6 @@ MONGODB_URI=
 # For using Pinecone (store)
 PINECONE_API_KEY=
 PINECONE_HOST=
+
+# Some examples are expensive to run, so we disable them by default
+RUN_EXPENSIVE_EXAMPLES=false
```
Comment (chr-hertel) [inline .env:29]: :D love it

File: examples/structured-output-math.php [hunk 2]
```diff
@@ -23,6 +23,11 @@
     exit(1);
 }
 
+if (empty($_ENV['RUN_EXPENSIVE_EXAMPLES']) || false === filter_var($_ENV['RUN_EXPENSIVE_EXAMPLES'], FILTER_VALIDATE_BOOLEAN)) {
+    echo 'This example is marked as expensive and will not run unless RUN_EXPENSIVE_EXAMPLES is set to true.'.PHP_EOL;
+    exit(1);
+}
+
```
Comment (chr-hertel) [inline examples/structured-output-math.php:30]: wrong example

File: .env (hunk 1 above)
Comment (chr-hertel) [reply .env:29]: maybe rather an option on the example runner?

File: examples/structured-output-math.php (hunk 2 above)
Comment (OskarStark) [reply examples/structured-output-math.php:30]: indeed

**Review by chr-hertel — APPROVED** [review]

File: .env (hunk 1 above)
Comment (OskarStark) [reply .env:29]: I think its safe here, at least I, run the example one by one via PHP instead of the runner

Comment (OskarStark) [reply .env:29]: > :D love it

I think those examples are key and the best documentation for the users

**Pull request merged**